tests/test_main.py

```diff
@@ -0,0 +1,371 @@
+
+
+class TestAuditUUIDModels(TestAuditModels):
+    @pytest.fixture
+    def model(self):
+        return UUIDModel
+
+    @pytest.fixture
+    def fk_model(self):
+        return UUIDForeignKeyModel
+
+    @pytest.fixture
+    def m2m_model(self):
+        return UUIDM2MModel
+
+
+class TestAuditBigIntModels(TestAuditModels):
+    @pytest.fixture
+    def model(self):
+        return BigIntModel
+
+    @pytest.fixture
+    def fk_model(self):
+        return BigIntForeignKeyModel
+
+    @pytest.fixture
+    def m2m_model(self):
+        return BigIntM2MModel
+
+
+@pytest.mark.django_db
+class TestMiddleware:
+    def test_middleware_logged_in(self, user, client, username, password):
+        client.login(username=username, password=password)
+        client.post(reverse("test_app:create-obj"))
+        assert Model.objects.count() == 1
+
+        obj = Model.objects.all().first()
+        crud_event = CRUDEvent.objects.filter(
+            object_id=obj.id, content_type=ContentType.objects.get_for_model(obj)
+        ).first()
+        assert crud_event.user == user
+
+    def test_middleware_not_logged_in(self, client):
+        create_obj_url = reverse("test_app:create-obj")
+        client.post(create_obj_url)
+        assert Model.objects.count() == 1
+
+        obj = Model.objects.all().first()
+        crud_event = CRUDEvent.objects.filter(
+            object_id=obj.id, content_type=ContentType.objects.get_for_model(obj)
+        ).first()
+        assert crud_event.user is None
+
+    def test_manual_set_user(self, django_user_model, username, email, password):
+        user = django_user_model.objects.create_user(username, email, password)
+        set_current_user(user)
+        obj = Model.objects.create()
+        assert obj.id == 1
+
+        crud_event_qs = CRUDEvent.objects.filter(
+            object_id=obj.id, content_type=ContentType.objects.get_for_model(obj)
+        )
+        assert crud_event_qs.count() == 1
+
+        crud_event = crud_event_qs.first()
+        assert crud_event.user == user
+
+        clear_request()
+        obj = Model.objects.create()
+        assert obj.id == 2
+
+        crud_event_qs = CRUDEvent.objects.filter(
+            object_id=obj.id, content_type=ContentType.objects.get_for_model(obj)
+        )
+        assert crud_event_qs.count() == 1
+
+        crud_event = crud_event_qs.first()
+        assert crud_event.user is None
+
+    def test_middleware_logged_in_user_in_request(self, user, client):
+        client.force_login(user)
+        create_obj_url = reverse("test_app:create-obj")
+        client.post(create_obj_url)
+        assert Model.objects.count() == 1
+
+        obj = Model.objects.first()
+        crud_event = CRUDEvent.objects.filter(
+            object_id=obj.id, content_type=ContentType.objects.get_for_model(obj)
+        ).first()
+        assert crud_event.user == user
+
+
+@pytest.mark.asyncio
+@pytest.mark.django_db(transaction=True)
```

Comment:
`transaction=True` makes this behave the same as if this class inhereted from `TransactionTestCase`